index.js

```diff
@@ -18,34 +16,25 @@ var typedArrays = {
 };
 
 var slice = String.prototype.slice;
-var toStrTags = {};
 var gOPD = Object.getOwnPropertyDescriptor;
-if (hasToStringTag && gOPD && Object.getPrototypeOf) {
-	forEach(typedArrays, function (_, typedArray) {
-		var arr = new global[typedArray]();
-		if (!(Symbol.toStringTag in arr)) {
-			throw new EvalError('this engine has support for Symbol.toStringTag, but ' + typedArray + ' does not have the property! Please report this.');
-		}
-		var proto = Object.getPrototypeOf(arr);
-		var descriptor = gOPD(proto, Symbol.toStringTag);
-		if (!descriptor) {
-			var superProto = Object.getPrototypeOf(proto);
-			descriptor = gOPD(superProto, Symbol.toStringTag);
-		}
-		toStrTags[typedArray] = descriptor.get;
-	});
-}
-
-var tryTypedArrays = function tryTypedArrays(value) {
-	var anyTrue = false;
-	forEach(toStrTags, function (getter, typedArray) {
-		if (!anyTrue) {
-			try {
-				anyTrue = getter.call(value) === typedArray;
-			} catch (e) { /**/ }
+var tryTypedArrays = function isTypedArray(value) {
+	var typedArray;
+	if (Object.getPrototypeOf) {
```

Comment:
since these are now evaluated at runtime, this is not robust against `delete Object.getPrototypeOf`, or `global.Symbol = {};`. Any value used at runtime must be evaluated and cached at module parse time.